### PR TITLE
Optimize database scheduled task

### PR DESCRIPTION
**Pull request by ferferga**
Before, we ran VACUUM at each start (#1455). Although it solved the problem partially, it isn't the best solution for long-running instances. It also freezes the boot process for a while and it can hang it altogether, in case the filesystem is running out of space.

With a scheduled task, this can be ran periodically at user demand, while also keeping server running in case of any error.

**I tested this while running scanning tasks and browsing the library with multiple users and I've never hit any concurrency issue**

### Additional details

* Although this might pose a concurrency problem, our current journaling mode and the fact that library is still in it's own database should not cause any issues **in case we want this for 10.7**.
* In addition to this, I've been testing vacuuming in different connections with my pi-hole instance multiple times this week: 

1. I replicated the pragmas used by Jellyfin.
2. Ran a DNS benchmark (so database got plenty of activity)
3. Connected using ``sqlite3`` cli and issued VACUUM command.

I tried multiple combinations of this and **nver had a single issue**. I read also a lot of the documentation and I got to the conclusion that, as long as there's **one connection that handles transactions** and **one connection that runs vacuum without a transaction**, everything is fine atomic-wise. Anyway, **I'm leaving library.db as it's going to be deprecated sooner than later and it's better to stay in the safe side**.

* I wanted to use EFCore for this but database context is not accesible outside Jellyfin namespace. Anyway, I don't think doing it directly with SQLitePCL is such a bad idea as, once support for other backends is added, we would need to set a check so this trigger only works in SQLite backends. With SQLitePCL we have this done already.

### CODE REVIEW

File: Emby.Server.Implementations/ScheduledTasks/Tasks/OptimizeDatabase.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        /// <param name="progress">The progress.</param>
+        /// <returns>Task.</returns>
+        public Task Execute(CancellationToken cancellationToken, IProgress<double> progress)
+        {
+            var dataPath = _paths.DataPath;
+            _logger.LogInformation("Issuing VACUUM command to jellyfin.db...");
+
+            try
+            {
+                using var connection = SQLite3.Open(Path.Combine(dataPath, "jellyfin.db"));
```
Comment (crobibero) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/OptimizeDatabase.cs:88]: Should also vacuum `library.db` and `authentication.db`, as those were what was vacuumed, not `jellyfin.db`

Comment (ferferga) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/OptimizeDatabase.cs:88]: I know but I don't think it makes much sense given that those databases are going to be deprecated sooner than later (once they're EFCore as well). 

What the rest of you guys think? @jellyfin/server

Comment (ggogel) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/OptimizeDatabase.cs:88]: Maybe you could add an option "de-fragment database on startup". So users, who have their database on suboptimal network storage, could disable this or only enable it once in a while if the database is getting too fragmented. Also a user might want to disable this simply to conserve SSD / flash drive durability. Personally I run a secondary dev/beta instance to test out new versions and sometimes deploy a couple of times a day to that instance. Writing 700 Mb every time the server starts adds up to a lot of wasted TBW pretty quickly. In my opinion such an option would please everybody.

Comment (ferferga) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/OptimizeDatabase.cs:88]: @ggogel While I understand your concerns, I don't think it's worth it for a number of reasons:

• As we already discussed in Matrix, this is an stopgap while we get a proper database schema. Once we have it, fully integrated with EFCore, we would be able to completely vacuum everything at runtime with an scheduled tasks, which users can configure. So the problem will not be there forever, will be solved at some point. **Imo, this is just an stopgap for another stopgap** and stopgaps are cool but they shouldn't be abused.

• The use case you're exposing is pushing SQLite's design to it's limits already, as it's not designed to run over networks. Same for Jellyfin and it's data. The argument still won't fix the already edge case.


• Modern SSDs aren't that affected by TBW.

•VACUUM actually reduces the file size of the writes.


• Documenting, testing + everything else this new setting... For how many people really? Vacuum at startup has been running for almost a year and you are the first to notice it :).

• Jellyfin is a long running task: as a server, most people have it running in the background. As a long-running task, startups are not frequent.

I also suspect that, given you're running everything over a network, database fragmentation is even worse, that's why you're also experiencing high IO as iirc you mentioned correctly.

TLDR: There's no point imo to add an stopgap for another stopgap which doesn't provide a benefit/solution to a good amount of users. Your usage case is at the edge already.

**Comment by ferferga** [comment]
Cherrypicked in #6031
